### PR TITLE
Parachain Release

### PR DESCRIPTION
**Pull request by Gauthamastro**
Pallets installed.

1. Thea executor
2. Thea message handler 
3. Thea council
4. XCM configuration.

This release contains the feature updates to release thea and XCM logic to production.

### CODE REVIEW

File: pallets/xcm-helper/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+			let deposit = Deposit {
+				recipient,
+				asset_id,
+				amount: 1_000_000_000_000_000u128,
+				extra: Vec::new(),
+			};
+			let network = T::ParachainNetworkId::get();
+			T::Executor::execute_withdrawals(network, deposit.encode()).unwrap();
+			Ok(())
+		}
```
Comment (serhii-temchenko) [inline pallets/xcm-helper/src/lib.rs:364]: I assume this supposed to be removed before releasing

File: pallets/xcm-helper/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+//! - `ActiveTheaKey` - Stores Latest Thea Key.
+//! - `WithdrawNonce` - Stores Latest withdrawal nonce.
+//! - `PendingWithdrawals` - Stores all pending withdrawal.
+//! - `FailedWithdrawals` - Stores failed withdrawals which failed during execution.
+//! - `TheaAssets` - Stores all Thea Assets.
+//! -
+//! # Events
+//! - `AssetDeposited` - Asset Deposited from XCM.
+//! - `AssetWithdrawn` - Asset burned/locked from native Parachain.
+//! - `TheaAssetCreated` - New Asset Created.
```
Comment (serhii-temchenko) [inline pallets/xcm-helper/src/lib.rs:83]: This documentation is outdated, multiple members from this doc were removed. Please update it.

File: xcm-simulator/src/parachain.rs [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+pub type Barrier = (
+	TakeWeightCredit,
+	AllowTopLevelPaidExecutionFrom<Everything>,
+	// Expected responses are OK.
+	AllowKnownQueryResponses<PolkadotXcm>,
+	// Subscriptions for version tracking are OK.
+	AllowSubscriptionsFrom<Everything>,
+);
+
+//TODO: move DenyThenTry to polkadot's xcm module.
```
Comment (serhii-temchenko) [inline xcm-simulator/src/parachain.rs:163]: Is it still required?

File: xcm-simulator/src/parachain.rs [hunk 4]
```diff
@@ -0,0 +1,768 @@
...
+			ParachainId::<T>::put(para_id);
+		}
+
+		fn handle_xcmp_message(
+			sender: ParaId,
+			_sent_at: RelayBlockNumber,
+			xcm: VersionedXcm<T::RuntimeCall>,
+			max_weight: Weight,
+		) -> Result<Weight, XcmError> {
+			//assert_eq!("hello", "no_hello");
```
Comment (serhii-temchenko) [inline xcm-simulator/src/parachain.rs:310]: Should be removed

File: xcm-simulator/src/parachain.rs [hunk 5]
```diff
@@ -0,0 +1,768 @@
...
+			max_weight
+		}
+	}
+
+	impl<T: Config> DmpMessageHandler for Pallet<T> {
+		fn handle_dmp_messages(
+			iter: impl Iterator<Item = (RelayBlockNumber, Vec<u8>)>,
+			limit: Weight,
+		) -> Weight {
+			//assert_eq!("hello", "no_hello");
```
Comment (serhii-temchenko) [inline xcm-simulator/src/parachain.rs:362]: Should be removed

File: xcm-simulator/src/parachain.rs [hunk 6]
```diff
@@ -0,0 +1,768 @@
...
+
+pub struct AccountIdToMultiLocation;
+impl Convert<AccountId, MultiLocation> for AccountIdToMultiLocation {
+	fn convert(account: AccountId) -> MultiLocation {
+		X1(Junction::AccountId32 { network: NetworkId::Any, id: account.into() }).into()
+	}
+}
+
+parameter_types! {
+	//pub ParaId: ParaId = mock_msg_queue::<Runtime>::get();
```
Comment (serhii-temchenko) [inline xcm-simulator/src/parachain.rs:492]: Should be removed

File: xcm-simulator/src/parachain.rs [hunk 7]
```diff
@@ -0,0 +1,768 @@
...
+	pub SelfLocation: MultiLocation = MultiLocation::new(1, X1(Parachain(MsgQueue::parachain_id().into()))); //TODO: CHnage to our Parachin Id
+	pub const BaseXcmWeight: XCMWeight = 100_000_000; // TODO: recheck this
```
Comment (serhii-temchenko) [inline xcm-simulator/src/parachain.rs:494]: Is these todo's still relevant?

File: pallets/xcm-helper/src/lib.rs (hunk 2 above)
Comment (serhii-temchenko) [reply pallets/xcm-helper/src/lib.rs]: Update could be found in the latest commits from me.

File: xcm-simulator/src/parachain.rs (hunk 6 above)
Comment (serhii-temchenko) [reply xcm-simulator/src/parachain.rs]: Removed

File: xcm-simulator/src/parachain.rs (hunk 4 above)
Comment (serhii-temchenko) [reply xcm-simulator/src/parachain.rs]: Removed

File: xcm-simulator/src/parachain.rs (hunk 5 above)
Comment (serhii-temchenko) [reply xcm-simulator/src/parachain.rs]: Removed

**Comment by serhii-temchenko** [comment]
Fixed part of my comments, waiting on some answers from @zktony.

File: pallets/xcm-helper/src/lib.rs (hunk 1 above)
Comment (zktony) [reply pallets/xcm-helper/src/lib.rs]: Yeah .. I will remove this

Comment (serhii-temchenko) [reply pallets/xcm-helper/src/lib.rs]: Removed in pr #47

File: xcm-simulator/src/parachain.rs (hunk 3 above)
Comment (Gauthamastro) [reply xcm-simulator/src/parachain.rs]: Removed, not our TODO

File: xcm-simulator/src/parachain.rs (hunk 7 above)
Comment (Gauthamastro) [reply xcm-simulator/src/parachain.rs:490]: @zktony What do you think?

File: pallets/xcm-helper/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,652 @@
...
+		/// Withdrawal Execution Failed
+		WithdrawalExecutionFailed,
+		/// Token Is Not Whitelisted
+		TokenIsNotWhitelisted,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(n: T::BlockNumber) -> Weight {
+			// TODO: Benchmark this is with a predefined bound but don't use bounded vec @zktony
```
Comment (serhii-temchenko) [inline pallets/xcm-helper/src/lib.rs:284]: Still remains before release @zktony

File: xcm-simulator/src/parachain.rs [hunk 9]
```diff
@@ -0,0 +1,764 @@
...
+pub struct AccountIdToMultiLocation;
+impl Convert<AccountId, MultiLocation> for AccountIdToMultiLocation {
+	fn convert(account: AccountId) -> MultiLocation {
+		X1(Junction::AccountId32 { network: NetworkId::Any, id: account.into() }).into()
+	}
+}
+
+parameter_types! {
+	pub SelfLocation: MultiLocation = MultiLocation::new(1, X1(Parachain(MsgQueue::parachain_id().into())));
+	pub const BaseXcmWeight: XCMWeight = 100_000_000; // TODO: recheck this @zktony
```
Comment (serhii-temchenko) [inline xcm-simulator/src/parachain.rs:490]: Still remains @zktony 
([Here](https://github.com/Polkadex-Substrate/parachain/pull/44#discussion_r1193440665) is an initial comment.)

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: xcm-simulator/src/parachain.rs (hunk 9 above)
Comment (Gauthamastro) [reply xcm-simulator/src/parachain.rs:490]: @zktony You mentioned this is fine, if so please remove the TODO

**Pull request merged**